libs/langchain-google-common/src/utils/gemini.ts

```diff
@@ -855,6 +858,57 @@ export function getGeminiAPI(config?: GeminiAPIConfig): GoogleAIAPI {
     };
   }
 
+  function addModalityCounts(
+    modalityTokenCounts: ModalityTokenCount[],
+    details: InputTokenDetails | OutputTokenDetails
+  ): void {
+    modalityTokenCounts?.forEach((modalityTokenCount) => {
+      const { modality, tokenCount } = modalityTokenCount;
+      const modalityLc = modality.toLowerCase();
+      const currentCount = details[modalityLc] ?? 0;
+      // eslint-disable-next-line no-param-reassign
+      details[modalityLc] = currentCount + tokenCount;
+    });
+  }
+
+  function responseToUsageMetadata(
+    response: GoogleLLMResponse
+  ): UsageMetadata | undefined {
+    if ("usageMetadata" in response.data) {
+      const data: GenerateContentResponseData = response?.data;
+      const usageMetadata = data?.usageMetadata;
+
+      const input_tokens = usageMetadata.promptTokenCount ?? 0;
+      const candidatesTokenCount = usageMetadata.candidatesTokenCount ?? 0;
+      const thoughtsTokenCount = usageMetadata.thoughtsTokenCount ?? 0;
+      const output_tokens = candidatesTokenCount + thoughtsTokenCount;
+      const total_tokens =
+        usageMetadata.totalTokenCount ?? input_tokens + output_tokens;
+
+      const input_token_details: InputTokenDetails = {};
+      addModalityCounts(usageMetadata.promptTokensDetails, input_token_details);
+
+      const output_token_details: OutputTokenDetails = {};
+      addModalityCounts(
+        usageMetadata?.candidatesTokensDetails,
+        output_token_details
+      );
+      if (usageMetadata?.thoughtsTokenCount) {
```

Comment:
```suggestion
      if (usageMetadata?.thoughtsTokenCount != null) {
```

else we'll miss zeros